### PR TITLE
ENH: support reading GeoParquet with native geoarrow-based encodings

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Broken off from https://github.com/geopandas/geopandas/pull/3275 with just the read support.

Implementing the reading part of https://github.com/geopandas/geopandas/issues/3253

### CODE REVIEW

**Review by martinfleis — APPROVED** [review]
Looks great! It just need a changelog note.

File: geopandas/io/geoarrow.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                [
+                    np.asarray(arr.field("x")),
+                    np.asarray(arr.field("y")),
+                    np.asarray(arr.field("z")),
+                ]
+            )
+        return coords
+    else:
+        # fixed size list
+        return np.asarray(arr.values).reshape(len(arr), -1)
```
Comment (m-richards) [inline geopandas/io/geoarrow.py:25]: Is it easy to generate an input file to read so our tests cover the interleaved coordinates case? (we also miss coverage above for the 2D/3D case, but the logical is identical so I don't think that matters.

**Review by m-richards — APPROVED** [review]
Looks fine to me as well, from my high level understanding of geoparquet/geoarrow

File: geopandas/io/geoarrow.py (hunk 1 above)
Comment (jorisvandenbossche) [reply geopandas/io/geoarrow.py:25]: So officially GeoParquet doesn't support interleaved coordinates, only separated. That's why it is not included here in the tests. In a follow-up PR, I however want to add support for converting from (Geo)Arrow (in-memory) to geopandas (https://github.com/geopandas/geopandas/pull/3219 is already doing the export from geopandas to geoarrow), and those code paths will be covered. 

Not ideal that for now it's not covered (and I could also have just included the separated code path here), but will do that follow-up shortly.

Comment (m-richards) [reply geopandas/io/geoarrow.py:25]: For reference, this was done in #3301

**Pull request merged**